internal/apis/certmanager/types_certificate.go

```diff
@@ -371,6 +371,14 @@ type CertificateStatus struct {
 	// It will automatically unset this field when the Issuing condition is
 	// not set or False.
 	NextPrivateKeySecretName *string
+
+	// The number of times certificate issuance has been attempted and
+	// failed for this revision (see the description of status.revision
+	// field). If an issuance has failed, the delay till the next issuance
+	// will be calculated using formula time.Hour * 2 ^ (issuanceAttempts - 1).
+	// This field will only be set if the issuance is currently failing.
+	// +optional
+	IssuanceAttempts *int `json:"issuanceAttempts,omitempty"`
```

Comment:
It was not clear to me when reading the name of this field whether this value relates to the _total_ number of issuance attempts for the entirety of this Certificate's lifetime, or if it only related to the current revision. Although the comments clear it up, maybe which should make the name of the field more clear? Something like `revisionIssuanceAttemptes`?

Reply:
> whether this value relates to the total number of issuance attempts for the entirety of this Certificate's lifetime, or if it only related to the current revision

It is neither of those two. It is the number of failed issuances since last successful issuance (i.e how many times _in a row_ the issuance has failed)
I tried to take revision out of this work as I find that it makes things more confusing.
This PR does not reset exponential backoff if cert's spec has changed (which would cause the revision number to be bumped although that wouldn't appear on the `Certificate`'s status unless there was a successful issuance). If the spec has changed that will cause an immediate attempt to re-issue but if we are already in backoff and that issuance fails, the failed attempt will count towards the existing backoff period).

Reply:
> I will change it to failedIssuanceAttempts unless there are better suggestions

That name makes sense to me :slightly_smiling_face:

Reply:
Yep, this seems to be the case :+1: